json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java

```diff
@@ -32,10 +42,69 @@ public void setUp() throws Exception {
 			LocalDateTime.of(2017, 1, 1, 13, 37),
 			Optional.of("foo"),
 			OptionalInt.of(15));
+
+
 	}
 
 	@Test
-	public void shouldSerializeAndDeserializeImmutableObjects() throws Exception {
+    public void shouldThrowInvalidJsonException() {
+        Function<String, ImmutableEntity> deserializer = deserializerFactory.createDeserializer(ImmutableEntity.class);
+
+        try {
+            deserializer.apply("not real json");
+            fail("Expected exception, but none was thrown");
+        } catch(Exception actualException) {
+            assertThat(actualException).isInstanceOf(InvalidJsonException.class);
+            assertThat(actualException.getMessage()).contains("Unrecognized token");
+        }
+    }
+
+    @Test
+    public void shouldDeserializeNullToNull() {
+        Function<String, ImmutableEntity> deserializer = deserializerFactory.createDeserializer(ImmutableEntity.class);
+
+        ImmutableEntity result = deserializer.apply(null);
+
+        assertThat(result).isNull();
+    }
+
+
+    @Test
+    public void shouldshouldSerializeAndDeserializeImmutableObjectsByTypeReference() {
```

Comment:
A superfluous should has snuck it's way into the functionname.

Reply:
Fixed!